R/viz.R

```diff
@@ -2354,18 +2356,23 @@ prettyForestPlot = function(maf,
       dplyr::rename("Tumor_Sample_Barcode"="sample_id")
   }else if(!missing(maf)){
     if(separate_hotspots){
-        if(!"hot_spot" %in% colnames(maf))
-          stop("No \"hot_spot\" column in maf file. Annotate your maf file with GAMBLR::annotate_hot_spots() first. ")
+      if(!"hot_spot" %in% colnames(maf))
+        stop("No \"hot_spot\" column in maf file. Annotate your maf file with GAMBLR::annotate_hot_spots() first. ")
       maf$Hugo_Symbol = ifelse(!is.na(maf$hot_spot), paste0(maf$Hugo_Symbol, "_hotspot"), maf$Hugo_Symbol)
     }
     #Convert the maf file to a binary matrix
     mutmat = maf %>%
       dplyr::select(Hugo_Symbol, Tumor_Sample_Barcode) %>%
-      left_join(dplyr::select(metadata, Tumor_Sample_Barcode, comparison), by = "Tumor_Sample_Barcode") %>%
+      full_join(dplyr::select(metadata, Tumor_Sample_Barcode, comparison), by = "Tumor_Sample_Barcode") %>%
       distinct() %>%
       dplyr::mutate(is_mutated = 1) %>%
       pivot_wider(names_from = Hugo_Symbol, values_from = is_mutated, values_fill = 0) %>%
       dplyr::mutate(across(where(is.numeric), ~replace_na(., 0)))
+    if(rm_na_samples){ #remove 0 mutation samples, i.e samples that show 0 for all selected genes.
+      names(mutmat)[length(names(mutmat))]<-"na_samples" #rename the last column, "NA" (since dplyr do not treat a variable named NA the same way as other variables?)
+      mutmat = mutmat %>%
+        dplyr::filter(na_samples == 0) %>% #filter out all samples that show no mutations in the selected genes (i.e na_samples = 1).
+        dplyr::select(-na_samples) #drop temporary column (na_samples) to return mutmat to expected format.
```

Comment:
If you put this step (`select(-na_samples)`) outside this conditional subsetting, you can be sure to drop the `na_samples` column regardless of the value of `rm_na_samples`

Reply:
Fixed in [this](https://github.com/morinlab/GAMBLR/pull/88/commits/ba5f3a8078f305d9cc1931d199bca22a0aa92c9b) commit

Reply:
Shall I approve this on my end?

Reply:
Yes, please.